### DISCUSSION TITLE
How to run containers on different machines on the same host? (on Mac)

### DISCUSSION THREAD

**Discussion by pavelanni** [containers/podman#13524] [Q&A] [answered]
The problem: I want to be able to run containers on different machines on the same host. For example, I created several machines with different parameters: disk size, mounted volumes, etc. I want to run one container _with_ access to my mounted volumes and other containers _without_ it. I would imagine an additional argument to container commands should exist, like: `podman container start --machine machine1` or `podman container init --machine machine2`.
For now, the only way to work with different machines I found is: `podman system connection default machine1` -- then this machine becomes the default and all container commands are applied to it.  But it doesn't look very convenient, I hope there is a better way. What am I missing?

**Comment by baude** [comment]
why do you need two machines to run one container with a volume and one without?  one container would have a -v macos:vm and the other wouldn't

**Reply by pavelanni** [reply]
Using different volumes was just one of the examples. Let's put it this way: if I can create more than one machine to run containers in them -- I should have an easy way to tell `podman` which machine I want to use for which container. Using `CONTAINER_CONNECTION` as @Luap99 suggested is a good way, but having a CLI argument in addition to that would be even better.

**Comment by Luap99** [comment, marked as answer]
You set the `CONTAINER_CONNECTION` environment variable, e.g. `CONTAINER_CONNECTION=machine2 podman ...`
Or you just run `export CONTAINER_CONNECTION=machine2` to set for you current session so you can use just `podman ...`

**Reply by pavelanni** [reply]
That works! Thanks, @Luap99!

**Reply by Luap99** [reply]
You can also use cli arg for this. `podman --connection <name> ...`

**Reply by pavelanni** [reply]
This is what I was looking for! Somehow I couldn't find it in the docs, but now I see it in `--help`.
I guess it's coming from "podman-remote" -- that's why it's `connection` and not `machine`.
Thanks again!

**Reply by Luap99** [reply]
Yes podman machine just manages the VM and sets up a connection. The rest is handled by the regular remote client which exists much longer than podman machine. https://docs.podman.io/en/latest/markdown/podman-remote.1.html

**Comment by alanx172-cn** [comment]
you can run like this:
podman -c machine1 run hello  # run in machine machine1
podman run hello  # run in default machine
